usl_models/usl_models/flood_ml/model.py

```diff
@@ -128,6 +134,60 @@ def call_n(self, full_input: Input, n: int = 1) -> tf.Tensor:
         """Predict the next n timesteps. See `FloodConvLSTM.call_n`."""
         return self._model.call_n(full_input, n=n)
 
+    def batch_predict_n(
+        self, dataset: tf.data.Dataset, n: int = 1
+    ) -> Iterator[list[Result]]:
+        """Runs batch prediction (call_n).
+
+        The strategy should be the same as the one used to initialize the model.
+
+        Example usage:
+        ```py
+        strategy = tf.distribute.MirroredStrategy()
+        with strategy.scope():
+          model = FloodModel.from_checkpoint(artifact_uri="gs://path/to/model")
+          for results in model.batch_predict_n(strategy, dataset, n=4):
+            for result in results:
+              print(result)
+        ```
+
+        Args:
+            strategy: multi-GPU distribute strategy.
+            dataset: Dataset generating (inputs, metadata) tuples.
+            n: number of timesteps to predict.
+
+        Returns: an iterator containing batches of results.
+        """
+        strategy = tf.distribute.get_strategy()
+        dataset = strategy.experimental_distribute_dataset(dataset)
```

Comment:
always exciting to see the word `experimental` in a method name :)
how... 'experimental' is this?

Reply:
The docs don't discourage it's usage, seems like it's be stable for the past few TF versions: https://www.tensorflow.org/api_docs/python/tf/distribute/Strategy#experimental_distribute_dataset